test/common/performance/sender/http_load_generator.go

```diff
@@ -63,20 +64,27 @@ func NewCloudEventsTargeter(sinkUrl string, msgSize uint, eventType string, even
 }
 
 func (cet CloudEventsTargeter) VegetaTargeter() vegeta.Targeter {
+	body := []byte("\"" + generateRandString(cet.msgSize) + "\"")
```

Comment:
Caveat: a channel implementation that compresses blocks of messages (like Kafka) may get better performance here than in real life because messages with identical payloads compress better. Is generating these strings per-message causing issues?

Reply:
Generating a random string takes time and that code is ran inside the vegeta goroutine pool, so i've found that it can slow down the sender. I'm not sure we are gaining any advantage in kafka channel implementation about it. What about making it configurable?

Reply:
Looking at it now, there is even a little bit too much casting `[]byte -> string -> []byte` :thinking: